services/web/server/src/simcore_service_webserver/meta_handlers.py

```diff
@@ -0,0 +1,277 @@
+                raise ValueError("No iteration tag found")
+
+            iterations.append((workcopy_id, iteration.iter_index))
+
+        except ValueError as err:
+            log.debug("Skipping child %s: %s", n, err)
+
+    total_number_of_iterations = len(iterations)
+
+    # sort and select. If requested interval is outside of range, it returns empty
+    iterations.sort(key=lambda tup: tup[1])
+
+    if limit is None:
+        return iterations[offset:], total_number_of_iterations
+
+    return iterations[offset : (offset + limit)], total_number_of_iterations
+
+
+async def create_or_get_project_iterations(
+    vc_repo: VersionControlForMetaModeling,
+    project_uuid: ProjectID,
+    commit_id: CommitID,
+) -> List[IterationTuple]:
+
+    raise NotImplementedError()
+
+
+# MODELS ------------------------------------------------------------
+
+
+class ParentMetaProjectRef(BaseModel):
+    project_id: ProjectID
+    ref_id: CheckpointID
+
+
+class ProjectIterationAsItem(BaseModel):
+    name: str = Field(
+        ...,
+        description="Iteration's resource name [AIP-122](https://google.aip.dev/122)",
+    )
+    parent: ParentMetaProjectRef = Field(
+        ..., description="Reference to the the meta-project that created this iteration"
+    )
+
+    workcopy_project_id: ProjectID = Field(
+        ...,
+        description="ID to this iteration's working copy."
+        "A working copy is a real project where this iteration is run",
+    )
+
+    workcopy_project_url: HttpUrl = Field(
+        ..., description="reference to working copy project"
+    )
+    url: HttpUrl = Field(..., description="self reference")
+
+
+# ROUTES ------------------------------------------------------------
+
+
+routes = web.RouteTableDef()
+
+
+@routes.get(
+    f"/{VTAG}/projects/{{project_uuid}}/checkpoint/{{ref_id}}/iterations",
+    name=f"{__name__}._list_meta_project_iterations_handler",
+)
+@permission_required("project.snapshot.read")
+async def _list_meta_project_iterations_handler(request: web.Request) -> web.Response:
+    # FIXME: check access to non owned projects user_id = request[RQT_USERID_KEY]
+
+    # parse and validate request ----
+    url_for = create_url_for_function(request)
+    vc_repo = VersionControlForMetaModeling(request)
+
+    _project_uuid = ProjectID(request.match_info["project_uuid"])
+    _ref_id = request.match_info["ref_id"]
+
+    _limit = int(request.query.get("limit", DEFAULT_NUMBER_OF_ITEMS_PER_PAGE))
+    _offset = int(request.query.get("offset", 0))
+
+    try:
+        commit_id = CommitID(_ref_id)
+    except ValueError as err:
+        # e.g. HEAD
+        raise NotImplementedError(
+            "cannot convert ref (e.g. HEAD) -> commit id"
+        ) from err
+
+    # core function ----
+    (
+        selected_project_iterations,
+        total_number_of_iterations,
+    ) = await list_project_iterations(
+        vc_repo, _project_uuid, commit_id, offset=_offset, limit=_limit
```

Comment:
using the NamedTuple here or dataclass would make this a little bit more compact.